packages/node-cli/src/commands/object/update/UpdateInputHandler.js

```diff
@@ -54,68 +102,69 @@ module.exports = class UpdateInputHandler extends BaseInputHandler {
 		if (foundXMLFiles.length === 0) {
 			throw NodeTranslationService.getMessage(ERRORS.NO_OBJECTS_IN_PROJECT);
 		}
+		return foundXMLFiles;
+	}
 
+	async getObjectsToSelect(foundXMLFiles) {
 		let filteredObjects;
 
 		if (foundXMLFiles.length > MAX_ENTRIES_BEFORE_FILTER) {
-			const filterAnswers = await prompt([
-				{
-					type: CommandUtils.INQUIRER_TYPES.LIST,
-					name: ANSWERS_NAMES.FILTER_BY_SCRIPT_ID,
-					message: NodeTranslationService.getMessage(QUESTIONS.FILTER_BY_SCRIPT_ID),
-					default: false,
-					choices: [
-						{ name: NodeTranslationService.getMessage(YES), value: true },
-						{ name: NodeTranslationService.getMessage(NO), value: false },
-					],
-				},
-				{
-					when: (response) => {
-						return response[ANSWERS_NAMES.FILTER_BY_SCRIPT_ID];
-					},
-					type: CommandUtils.INQUIRER_TYPES.INPUT,
-					name: ANSWERS_NAMES.SCRIPT_ID_FILTER,
-					message: NodeTranslationService.getMessage(QUESTIONS.SCRIPT_ID_FILTER),
-					validate: (fieldValue) => showValidationResults(fieldValue, validateScriptId),
-				},
-			]);
-			filteredObjects = filterAnswers[ANSWERS_NAMES.FILTER_BY_SCRIPT_ID]
-				? foundXMLFiles.filter((element) => element.value.includes(filterAnswers[ANSWERS_NAMES.SCRIPT_ID_FILTER]))
-				: foundXMLFiles;
+			filteredObjects = await this.filterObjectsByScriptId(filteredObjects, foundXMLFiles);
 			if (filteredObjects.length === 0) {
 				throw NodeTranslationService.getMessage(MESSAGES.NO_OBJECTS_WITH_SCRIPT_ID_FILTER);
 			}
 		} else {
 			filteredObjects = foundXMLFiles;
 		}
-
 		filteredObjects.push(new Separator());
 
-		const answers = await prompt([
-			{
-				when: foundXMLFiles.length > 1,
-				type: CommandUtils.INQUIRER_TYPES.CHECKBOX,
-				name: ANSWERS_NAMES.SCRIPT_ID_LIST,
-				message: NodeTranslationService.getMessage(QUESTIONS.SCRIPT_ID),
-				default: 1,
-				choices: filteredObjects,
-				validate: (fieldValue) => showValidationResults(fieldValue, validateArrayIsNotEmpty),
-			},
+		return filteredObjects;
```

Comment:
I think we shouldn't add the new Separator() to the filteredObjects. And later in the next method selectObjectsScriptIds we could check if the length of the filteredObjects is bigger 1 to actually ask about which objects the user want to update.
Actually after some tryout I have found that with the current implementation we are not skipping the question even when there is only one object after the filtering, which may be good so the user will see the object that is being updated, but if we go in this direction I think we should remove the when clause on line 126. Actually I think that if there is only one object in the project the code will fail because the `return answers[ANSWERS_NAMES.SCRIPT_ID_LIST];` will be undefined.